### PR TITLE
fix(terminal): fix stack overflow in libvterm

### LINKED ISSUE #16040: Neovim 0.5.0 segfaults on chansend to terminal buffer opened with nvim_open_term
### Neovim version (nvim -v)

0.5.0

### Vim (not Nvim) behaves the same?

N/A

### Operating system/version

On both macOS 11.5.1 and CentOS 7.9

### Terminal name/version

On both iterm2 3.4.10 and xfce4-terminal 0.8.7.4

### $TERM environment variable

tmux-256color and xterm-256color (repros outside tmux)

### Installation

system package manager

### How to reproduce the issue

Save the following as `/tmp/crash.vim`:

```
let chan = nvim_open_term(0, {})
let lines = repeat([repeat('x',187)."\r"], 12250)
call chansend(chan, lines)
```

Then use `nvim --clean -S /tmp/crash.vim`.

### Expected behavior

terminal buffer opens with lines of x's

### Actual behavior

segfault

### PR DESCRIPTION
**Pull request by erw7**
- [x] write test.

fixes #16040.

### CODE REVIEW

File: src/nvim/terminal.c [hunk 1]
```diff
@@ -701,7 +701,17 @@ void terminal_receive(Terminal *term, char *data, size_t len)
     return;
   }
 
-  vterm_input_write(term->vt, data, len);
+  // Becatuse libvterm uses VLA, it casues stack overflow when large data is
```
Comment (jvoisin) [inline src/nvim/terminal.c:704]: `s/Becatuse/Because` :)

Comment (jvoisin) [reply src/nvim/terminal.c:704]: `s/casues/causes`

File: src/nvim/terminal.c [hunk 2]
```diff
@@ -701,7 +701,17 @@ void terminal_receive(Terminal *term, char *data, size_t len)
...
+  // written at once. To prevent this, the data will be split and written until
+  // libvterm is fixed.
```
Comment (jvoisin) [inline src/nvim/terminal.c:706]: Do you have a link to the relevant bug?

Comment (erw7) [reply src/nvim/terminal.c]: No one has reported a bug in [launchpad](https://bugs.launchpad.net/libvterm) yet. I think the only link to this bug is #16040.

**Pull request closed without merging**